src/anal_ghidra.cpp

```diff
@@ -0,0 +1,2405 @@
+			}
+
+			if(group == r_reg_type_arr[i])
+			{
+				buf << r_reg_string_arr[i] << '\t';
+				break;
+			}
+		}
+
+		buf << p->name << "\t." << p->size * 8 << "\t" << p->offset << "\t"
+		    << "0\n";
+	}
+	const std::string &res = buf.str();
+	// fprintf(stderr, res.c_str());
+	return strdup(res.c_str());
+}
+
+static bool sleigh_esil_consts_pick(RAnalEsil *esil)
+{
+	if(!esil || !esil->stack)
+		return false;
+
+	char *idx = r_anal_esil_pop(esil);
+	ut64 i;
+	int ret = false;
+
+	if(R_ANAL_ESIL_PARM_REG == r_anal_esil_get_parm_type(esil, idx))
+	{
+		throw LowlevelError("sleigh_esil_consts_pick: argument is consts only");
+		goto end;
+	}
+	if(!idx || !r_anal_esil_get_parm(esil, idx, &i))
+	{
+		throw LowlevelError("esil_pick: invalid index number");
+		goto end;
+	}
+	if(esil->stackptr < i)
+	{
+		throw LowlevelError("esil_pick: index out of stack bounds");
+		goto end;
+	}
+	if(!esil->stack[esil->stackptr - i])
+	{
+		throw LowlevelError("esil_pick: undefined element");
+		goto end;
+	}
+	if(!r_anal_esil_push(esil, esil->stack[esil->stackptr - i]))
+	{
+		throw LowlevelError("ESIL stack is full");
+		esil->trap = 1;
+		esil->trap_code = 1;
+		goto end;
+	}
+	ret = true;
+end:
+	free(idx);
+	return ret;
+}
+
+constexpr int ESIL_PARM_FLOAT = 127; // Avoid conflict
+
+static bool esil_pushnum_float(RAnalEsil *esil, long double num)
+{
+	char str[64];
+	snprintf(str, sizeof(str) - 1, "%.*LeF\n", DECIMAL_DIG, num);
+	return r_anal_esil_push(esil, str);
+}
+
+static int esil_get_parm_type_float(RAnalEsil *esil, const char *str)
+{
+	int len, i;
+
+	if(!str || !(len = strlen(str)))
+		return R_ANAL_ESIL_PARM_INVALID;
+
+	if((str[len - 1] == 'F') && (str[1] == '.' || (str[2] == '.' && str[0] == '-')))
+		return ESIL_PARM_FLOAT;
+	if(!strcmp(str, "nan") || !strcmp(str, "inf"))
+		return ESIL_PARM_FLOAT;
+	if(!((IS_DIGIT(str[0])) || str[0] == '-'))
+		goto not_a_number;
+
+	return ESIL_PARM_FLOAT;
+not_a_number:
+	if(r_reg_get(esil->anal->reg, str, -1))
+		return R_ANAL_ESIL_PARM_REG;
+
+	return R_ANAL_ESIL_PARM_INVALID;
+}
+
+static int esil_get_parm_float(RAnalEsil *esil, const char *str, long double *num)
+{
+	if(!str || !*str)
+		return false;
+
+	int parm_type = esil_get_parm_type_float(esil, str);
+	if(!num || !esil)
+		return false;
+
+	switch(parm_type)
+	{
+		case ESIL_PARM_FLOAT:
+			// *num = r_num_get (NULL, str);
+			scanf("%LfF", num);
+			return true;
+		case R_ANAL_ESIL_PARM_REG:
+			if(!r_anal_esil_reg_read(esil, str, (ut64 *)num, nullptr))
+			{
+				break;
+			}
+			return true;
+		default:
+			if(esil->verbose)
+			{
+				eprintf("Invalid arg (%s)\n", str);
+			}
+			esil->parse_stop = 1;
+			break;
+	}
+	return false;
+}
+
+static bool sleigh_esil_is_nan(RAnalEsil *esil)
+{
+	bool ret = false;
+	long double s;
+	char *src = r_anal_esil_pop(esil);
+	if(src && esil_get_parm_float(esil, src, &s))
+		ret = r_anal_esil_pushnum(esil, isnan(s));
+	else
+		throw LowlevelError("sleigh_esil_is_nan: invalid parameters");
+
+	free(src);
```

Comment:
These frees on memory malloced in r2 can become problematic depending on the C++ compiler/lib. Better use `r_mem_free()` for those.